upgrade/upgrade-scripts/upgrade

```diff
@@ -147,41 +159,45 @@ function upgrade_in_place() {
 	cleanup_in_place_upgrade
 	trap - EXIT
 
-	ROOTFS_CONTAINER="$(get_mounted_rootfs_container_name)"
-	[[ -n "$ROOTFS_CONTAINER" ]] ||
-		die "unable to determine currently mounted rootfs container"
+	if compare_versions "$CURRENT_VERSION" "ge" "12.0.0.0-0"; then
+		execute-in-place-upgrade -p
```

Comment:
What's the reason to not ship this new script `execute-in-place-upgrade` as part of the upgrade image itself?

Reply:
good question..

I think the downside of if this script was in the upgrade image, is then, in order to execute it, you'd need to know which image/directory to use..

E.g. I was thinking we'd eventually have a new service that could run `execute-in-place-upgrade -p`, and be able to restart if needed (e.g. automatically run after a kernel panic, or OOM kill, or something like that).. if this service needed to execute a script from the unpacked upgrade image, it'd need to know which unpacked upgrade image to use (since we could have multiple unpacked at a time)..

So, I think we could do it that way.. but we'd still need some logic outside of the upgrade image, that we use to determine which image to use, when executing the script.. does that make sense?

And since most of this script is just determining which upgrade image to choose (e.g. most of the options are for choosing the image to use; `-l`, `-i`, `-v`, and `-p`), and then running some scripts from that image (i.e. `execute`, `rootfs-container`, etc), I chose to just have the whole script live outside of the upgrade image..

I'm not necessarily opposed to moving this into the upgrade scripts, though.. I think that's potentially a better solution.. and then, most of the script probably goes away.. and the yet-to-be-created upgrade service that'd run the script, could still "parse" the `upgrade.properties` file to determine which image to use when running the script..

Reply:
Before your changes with these PRs, the `execute` script was always the single point of entry into actually performing the upgrade. The `execute` script was already shipped as part of the upgrade image. Would you consider extending the `execute` script to do all of the things that your new script is doing?
I think it's just simpler, solves the same problems you are trying to solve with these changes. I don't know if you have already thought of what this future service is supposed to do, but I'd presume it would simply be a thin wrapper to just run the `execute` script from the correct image....we could tell the service to use the upgrade properties API/file to determine the correct upgrade image.

Reply:
> Would you consider extending the execute script to do all of the things that your new script is doing?

Yea, that's what I was thinking as well, after thinking about your original question..

I just pushed another commit to try and do this.. let me know what you think.

With this update, I think I'll drop the `delphix-platform` change, and then the `dlpx-app-gate` change will need to be modified, to run `execute` instead of `execute-in-place-upgrade` (but still use `systemd-run`)..

> we could tell the service to use the upgrade properties API/file to determine the correct upgrade image

Yea.. so once we do the work to add the new service, I think this service could use:
```
EnvironmentFile=/var/dlpx-update/upgrade.properties
```
in its service definition.. and since we're now setting `UPGRADE_VERSION` in the file (with the changes in this PR, and the `dlpx-app-gate` PR), the service can use that field to know which `execute` script to run (i.e. which unpacked image to use).

Does that seem reasonable to you?

Reply:
Yeah, that sounds great.

Reply:
Something we need to consider now, though.. is we'll have an "old" virtualization service calling the "new" execute script.. so, I'll have to verify this actually works correctly..

E.g. we'll have the existing `UpgradeManagerImpl` logic calling the new `execute` script.. where the existing virtualization logic does not expect `execute` to perform the reboot or restart the services.. I need to think about if this is going to cause any problems..